core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/MetricsServiceImpl.java

```diff
@@ -395,15 +395,20 @@ public void setDateTimeService(DateTimeService dateTimeService) {
         return dataAccess.findAllTenantIds()
                 .flatMap(Observable::from)
                 .map(row -> row.getString(0))
-                .flatMap(dataAccess::findTenant)
-                .flatMap(Observable::from)
-                .map(Functions::getTenant);
+                .distinct()
```

Comment:
No, it's calling equals on the String methods (couple lines above we do map(row -> row.getString(0)), which is the tenantId. There's a test in the MetricsServiceImpl that checks that no duplicate lines are returned.